src/coreclr/tools/Common/TypeSystem/Common/ExplicitLayoutValidator.cs

```diff
@@ -109,27 +110,27 @@ private void AddToFieldLayout(int offset, TypeDesc fieldType)
                 {
                     if (offset % _pointerSize != 0)
                     {
-                        // Misaligned struct with GC pointers
+                        // Misaligned struct with GC pointers or ByRef
```

Comment:
Now that we take this code path for byref-like types, the above `if (fieldType.IsByRefLike && offset % _pointerSize != 0)` seems unnecessary.